src/modules/drawing/components/DrawingToolbox.vue

```diff
@@ -67,6 +76,18 @@ function onCloseClearConfirmation(confirmed) {
         store.dispatch('setDrawingMode', { mode: null, ...dispatcher })
     }
 }
+
+async function onSaveName() {
+    isActiveLayerNameEditMode.value = !isActiveLayerNameEditMode.value
+    if (activeKmlLayerName.value === activeKmlLayer.value.name) return
+    store.dispatch('updateLayer', {
+        index: activeKmlLayerIndex.value,
+        layer: { ...activeKmlLayer.value, name: activeKmlLayerName.value },
```

Comment:
I'm not sure a spread will be sufficient, as there is nested objects (such as the KML metadata).
If you want a short handed way of dealing with a targeted update, you can use the `updateLayers` action in this way
```Javascript
store.dispatch('updateLayers', {
    layers: [
        {
            id: activeKmlLayer.value.id,
            name: activeKmlLayerName.value,
        }
    ]
})
```